pkg/providers/tinkerbell/validator.go

```diff
@@ -283,3 +295,21 @@ func hardwareMap(hardwareList []*tinkhardware.Hardware) map[string]*tinkhardware
 
 	return hardwareMap
 }
+
+// getWorkflowMap returns all the workflows on the tinkerbell stack in the form of map with mac address as a key
+func getWorkflowMap(workflowList []*tinkworkflow.Workflow) (map[string]*tinkworkflow.Workflow, error) {
+	workflowMap := make(map[string]*tinkworkflow.Workflow)
+
+	for _, data := range workflowList {
+		var macAddress map[string]string
```

Comment:
nit: maybe call it `macAddressMap` or `macAddresses`
```suggestion
		var macAddressMap map[string]string
```

Reply:
Agree, I will change the name in my next PR as this one is merged. Thanks!